tokio/src/io/registration.rs

```diff
@@ -132,8 +132,19 @@ impl Registration {
 cfg_io_readiness! {
     impl Registration {
         pub(super) async fn readiness(&self, interest: mio::Interest) -> io::Result<ReadyEvent> {
-            // TODO: does this need to return a `Result`?
```

Comment:
It turns out that it does need to return a result :)

This implementation is not *great* but I believe the shutdown logic needs to be revisited post 0.3.

Reply:
I tracked my thoughts in #2924